pkgs/unified_analytics/lib/src/survey_handler.dart

```diff
@@ -0,0 +1,365 @@
+  Map<String, PersistedSurvey> fetchPersistedSurveys() {
+    final contents = _parseJsonFile();
+
+    // Initialize the list of persisted surveys and add to them
+    // as they are being parsed
+    var persistedSurveys = <String, PersistedSurvey>{};
+    contents.forEach((key, value) {
+      value as Map<String, dynamic>;
+
+      final uniqueId = key;
+      final snoozed = value['status'] == 'snoozed' ? true : false;
+      final timestamp =
+          DateTime.fromMillisecondsSinceEpoch(value['timestamp'] as int);
+
+      persistedSurveys[uniqueId] = PersistedSurvey(
+        uniqueId: uniqueId,
+        snoozed: snoozed,
+        timestamp: timestamp,
+      );
+    });
+
+    return persistedSurveys;
+  }
+
+  /// Retrieves the survey metadata file from [kContextualSurveyUrl]
+  Future<List<Survey>> fetchSurveyList() async {
+    final List<dynamic> body;
+    try {
+      final payload = await _fetchContents();
+      body = jsonDecode(payload) as List<dynamic>;
+      // ignore: avoid_catches_without_on_clauses
+    } catch (err) {
+      return [];
+    }
+
+    final surveyList = parseSurveysFromJson(body);
+
+    return surveyList;
+  }
+
+  /// Fetches the json in string form from the remote location
+  Future<String> _fetchContents() async {
+    final uri = Uri.parse(kContextualSurveyUrl);
+    final response = await http.get(uri);
+    return response.body;
+  }
+
+  /// Method to return a Map representation of the json persisted file
+  Map<String, dynamic> _parseJsonFile() {
+    Map<String, dynamic> contents;
+    try {
+      contents = jsonDecode(_dismissedSurveyFile.readAsStringSync())
+          as Map<String, dynamic>;
+    } on FormatException {
+      Initializer.createDismissedSurveyFile(
+          dismissedSurveyFile: _dismissedSurveyFile);
+      contents = {};
+    } on FileSystemException {
+      Initializer.createDismissedSurveyFile(
+          dismissedSurveyFile: _dismissedSurveyFile);
+      contents = {};
+    }
+
+    return contents;
+  }
+
+  /// Function to ensure that each survey is still valid by
+  /// checking the [Survey.startDate] and [Survey.endDate]
+  /// against the current [clock.now()] date
+  static bool checkSurveyDate(Survey survey) {
+    if (survey.startDate.isBefore(clock.now()) &&
+        survey.endDate.isAfter(clock.now())) return true;
+
+    return false;
+  }
+
+  /// Function that takes in a json data structure that is in
+  /// the form of a list and returns a list of [Survey]s
+  ///
+  /// This will also check the survey's dates to make sure it
+  /// has not expired
+  static List<Survey> parseSurveysFromJson(List<dynamic> body) => body
+      .map((element) {
+        // Error handling to skip any surveys from the remote location
+        // that fail to parse
+        try {
+          return Survey.fromJson(element as Map<String, dynamic>);
+          // ignore: avoid_catches_without_on_clauses
```

Comment:
When I try to do the following

```dart
        try {
          element as Map<String, dynamic>;
          element.remove('startDate');
          return Survey.fromJson(element);
        } on TypeError {
          return null;
        }
```

I get the following lint message
```
The type 'TypeError' should not be caught
because it is a subclass of 'Error'. (avoid_catching_errors)

(https://dart.dev/lints/avoid_catching_errors)
```

What's the best approach for catching type errors? I can add the `// ignore: avoid_catching_errors` line to prevent this

Reply:
Yeah, I think it's better to ignore that lint than to catch everything.